### ISSUE TITLE
Timing Bug: Wait commands and event movement off by 1 frame

### ISSUE THREAD

**Issue by fmatthew5876** [EasyRPG/Player#1623]
Create a map event with trigger = parallel and `anim_type = continuous`

Set this code
```
WAIT
OpenSaveMenu
```

Chunks of the event with different WAIT's at the time you save

| WAIT | RPG_RT stop count | Player stop count | RPG_RT anim_count | Player anim_count | RPG_RT anim frame | Player anim frame |
| -- | -- | -- | -- | -- | -- | -- |
| | 1 | 1 | 1 | 1 | 1 | 1 |
| 0.0s | 2 | 1 | 2 | 1 | 1 | 1 |
| 0.0s, 0.0s | 3 | 2 | 3 | 1 | 2 | 1 |
| 0.1s | 7 | 6 | 1 | 2 | 0 | 2 |
| 1.0s | 61 | 60 | 1 | 3 | 0 | 3

So there is an off by 1 error somewhere as shown in `stop_count` and `anim_count`. Either we're really waiting -1 frames or the order of updates with when the wait starts/ends is wrong somehow.

The `anim_count` results were taken with #1601 as master has these completely wrong to begin with.

**Comment by fmatthew5876** [comment]
This bug looks like it only happens when the first thing you do in the event code is wait.

For example, the below code works correctly
```
OpenSaveMenu
WAIT
OpenSaveMenu
```

**Comment by fmatthew5876** [comment]
For further investigation, I made a test with the code

```
Show Picture
Loop: 
  Wait
```

| Event Type | Result |
| -- | -- |
| Parallel Map Event | Picture appears before transition in |
| Autostart Map Event | Picture appears after transition in |
| Common parallel | Picture appears before transition in |
| Autostart common | Picture appears after transition in |

Now I change the code to this:

```
OpenSaveMenu
```


| Event Type | Player stop count | Event stop count |
| -- | -- | -- |
| Parallel Map Event | 0 | 1 |
| Autostart Map Event | 1 | 1 |
| Common parallel | 0 | 0 |
| Autostart common | 0 | 1 |

**Comment by fmatthew5876** [comment]
Multiple events:

Here is a more revealing test:

1. Create 2 events with code

```
OpenSaveMenu
Loop:
  wait
```

| EV1 Trigger | EV2 Trigger | EV1 stop count | EV2 stop count | EV1 anim count | EV2 anim count |
| -- | -- | -- | -- | -- | -- |
| | autorun | 2 | 1 | 2 | 2 |
| | parallel | 1 | 1 | 1 | 1 |
| autorun | 1 | 2 | 2 | 2 |
| parallel | 1 | 1 | 1 | 1 |

**Comment by fmatthew5876** [comment]
This is fixed by #1628, except the very last test case. But thats more than covered by #1606 cases so I consider this fixed when #1628 is merged.
